sufia-models/app/services/sufia/generic_file_indexing_service.rb

```diff
@@ -13,6 +13,7 @@ def generate_solr_document
         # between files on disk (in fcrepo.binary-store-path) and objects
         # in the repository.
         solr_doc[Solrizer.solr_name('digest', :symbol)] = digest_from_content
+        object.index_collection_ids(solr_doc) unless Sufia.config.collection_facet.nil?
```

Comment:
Not sure how this would become part of the background job.  The background job calls update_index which ends up calling the indexing service which calls object.index_collection_ids.  The background job itself does not have access to the solr_doc.  Do you have a suggestion for how to do this differently?